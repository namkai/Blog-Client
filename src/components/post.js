import React from 'react';

const Post = ({ _id, title, image, body, description, author, selectPost }) => (
	<li className="media list-group-item p-4" onClick={() => selectPost(_id)}>
		<img
			className="media-object d-flex align-self-start mr-3"
			src={image} />
		<div className="media-body">
			<div className="media-body-text">
				<div className="media-heading">
					<small className="float-right text-muted">{ author.name }</small>
					<h6>{title}</h6>
				</div>
				<p>
					{description}
				</p>
			</div>
		</div>
	</li>
);

export default Post;
