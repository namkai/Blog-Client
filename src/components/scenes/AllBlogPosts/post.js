import React from 'react';
import { Link } from 'react-router-dom';

//import '../../../style/post.css';

const Post = ({ _id, title, image, body, description, author, selectPost }) => (
	<li className="media list-group-item p-4 post" onClick={() => selectPost(_id)}>
		<img
			className="media-object d-flex align-self-start mr-3"
			src={image}/>
		<div className="media-body">
			<div className="media-body-text" style={{ width: '100%' }}>
				<div className="media-heading">
					<small className="float-right text-muted">{ author.name }</small>
					<Link to={`posts/${_id}`} onClick={() => selectPost(_id)}><h6>{title}</h6></Link>
				</div>
				<p>
					{description}
				</p>
			</div>
		</div>

	</li>
);

export default Post;
