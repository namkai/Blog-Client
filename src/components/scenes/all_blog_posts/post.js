import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

// import '../../../style/post.css';

export const Post = ({ _id, title, image, description, author, selectPost }) => (
	<li className="media list-group-item p-4 post" onClick={() => selectPost(_id)}>
		<img
			className="media-object d-flex align-self-start mr-3"
			src={image}
		/>
		<div className="media-body">
			<div className="media-body-text" style={{ width: '100%' }}>
				<div className="media-heading">
					<small className="float-right text-muted">{ author.name }</small>
					<Link to={`posts/${_id}`} onClick={() => selectPost(_id)}><h6>{title}</h6></Link>
				</div>
				<p>
					{description}
				</p>
			</div>
		</div>
	</li>
);

Post.defaultProps = {
	_id: PropTypes.string,
	title: '',
	image: 'http://www.bolton-associates.co.uk/wp-content/uploads/2015/10/analytics-programming.png',
	description: 'This is the description',
	author: {},
	selectPost: PropTypes.func,
};

Post.propTypes = {
	_id: PropTypes.string,
	title: PropTypes.string,
	image: PropTypes.string,
	description: PropTypes.string,
	author: PropTypes.object,
	selectPost: PropTypes.func,
};
export default Post;
