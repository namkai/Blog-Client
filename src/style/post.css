.post {
	box-shadow: 0 1px 2px rgba(0,0,0,0.15);
	transition: box-shadow 0.3s ease-in-out;
	/*width: 320px;*/
	/*margin: 10px auto;*/
}
.post:hover {
	box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
	z-index: 10;
}