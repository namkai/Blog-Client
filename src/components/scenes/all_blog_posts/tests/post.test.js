import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Post } from '../post';

it('renders correctly', () => {
	const post = shallow(
		<Post />
	);
	expect(toJson(post)).toMatchSnapshot();
});