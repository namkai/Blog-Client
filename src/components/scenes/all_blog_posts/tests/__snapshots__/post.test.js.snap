// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="media list-group-item p-4 post"
  onClick={[Function]}
>
  <img
    className="media-object d-flex align-self-start mr-3"
    src="http://www.bolton-associates.co.uk/wp-content/uploads/2015/10/analytics-programming.png"
  />
  <div
    className="media-body"
  >
    <div
      className="media-body-text"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="media-heading"
      >
        <small
          className="float-right text-muted"
        />
        <Link
          onClick={[Function]}
          replace={false}
          to="posts/function () { [native code] }"
        >
          <h6 />
        </Link>
      </div>
      <p>
        This is the description
      </p>
    </div>
  </div>
</li>
`;
